### ISSUE TITLE
Consider merging `fregante/github-issue-link-status` into refined github

### ISSUE THREAD

**Issue by tonglil** [refined-github/refined-github#7504]
Hello, I am following up on https://github.com/refined-github/refined-github/pull/7444 (which has been working great) and now wanting to do the same for https://github.com/fregante/github-issue-link-status, but I noticed it's now owned by you as well.

Would it be possible to merge that feature into refined github, instead of maintaining two separate extension now?

Thanks for your consideration.

**Comment by kidonng** [comment]
I would like to see this. Actual JS + CSS is ~200 lines and from my brief look we can almost cut that in half if moving into RGH.

Potential issues I see are two:

- [Rule 2](https://github.com/refined-github/refined-github/wiki/%22Can-you-add-this-feature%3F%22#2-it-makes-sense-as-a-default-for-everyone): The feature once added will be very prominent, we need to decide whether it really makes sense for _everyone_.
- [Rule 6](https://github.com/refined-github/refined-github/wiki/%22Can-you-add-this-feature%3F%22#6-it-doesnt-require-too-much-http-traffic): Issue/PR links are common so it is likely going to trigger on every conversation page. Can it be slow or eat up quota quickly? (though the latter is less a concern unless you are browsing new pages at light speed).

**Comment by tonglil** [comment]
Rule 2: fair consideration
Rule 6: that doesn't seem to change since I am already using the extension right now?

**Comment by kidonng** [comment]
> Rule 6: that doesn't seem to change since I am already using the extension right now?

Yeah it's not a blocker, just a reminder. We seldom worry about this kind of stuff but things do add up.

**Comment by fregante** [comment]
My concerns are:

- that feature is too _bright_, we should not color the entire link but just add the icons. I wanted to add multiple styles to the extension for a long time
- the feature adds an icon next to every link on the page, causing a layout shift (CLS). This should happen as early as possible to avoid causing that. Our observer could take care of that

I'm not particularly concerned about quota specifically because the feature would only work on GraphQL, which requires a token, which means our usage is wellll below the limits.



> can almost cut that in half if moving into RGH.

You're right, with the observer and [batched-function](https://github.com/Richienb/batched-function) this is probably not that big. It should look something like:

```js
observe(the selector, initLink);

function initLink(link) {
  link.append(PR or issue icon, depending on link.href)
  fetchInfo(link)
}

const fetchInfo = batchedFunction((links) => {
  // build query
  // query api
  // update links
});
```

**Comment by offbyone** [comment]
Does the fact that the extension has now stopped working on the Chrome store have any effect on the decision in this regard? I discovered it only this week, and am unable to install it. :D

**Comment by fregante** [comment]
Unfortunately due to the amount of bugs and my availability this is unlikely to see the light anytime soon. It's best someone sends a PR to update Parcel and migrate the extension to MV3 instead.

**Comment by fregante** [comment]
@g1eny0ung submitted a PR to upgrade the github-issue-link-status extension to MV3 so it will be live on the Chrome Web Store soon.
